### PR TITLE
Add `@mem` package

### PR DESCRIPTION
**Pull request by fantix**
This PR adds int/int64 load/store with little/big endian, as well as `@mem.copy()` to memory-like objects.

Ideally, such operations should use appropriate intrinsics.

Depends on:

- [ ] #328

### CODE REVIEW

File: mem/mem_test.mbt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+test "load bytes" {
+  test_load(Bytes::[1, 2, 3, 4, 5, 6, 7, 8, 9])?
+}
+
+test "load bytes view" {
+  test_load(
+    {
+      Bytes::[0, 1, 2, 3, 4, 5, 6, 7, 8, 9][1..]
+    },
```
Comment (fantix) [inline mem/mem_test.mbt:31]: `moon fmt` added unnecessary `{}` here.

Comment (peter-jerry-ye) [reply mem/mem_test.mbt:31]: cc @Yoorkin

Comment (Yoorkin) [reply mem/mem_test.mbt:31]: Fixed. The fix will apply in the next release.

**Comment by bobzhang** [comment]
you are welcome to revive this when you have time, close for now since it is inactive for a while.

**Pull request closed without merging**